transport/transport-runtime/src/main/java/com/google/android/datatransport/runtime/scheduling/jobscheduling/AlarmManagerScheduler.java

```diff
@@ -31,13 +38,65 @@
 
   private final Clock clock;
 
+  private AlarmManager alarmManager;
+
+  private final int DELTA = 30000; // 30 seconds delta
+
   @Inject
   public AlarmManagerScheduler(Context applicationContext, EventStore eventStore, Clock clock) {
     this.context = applicationContext;
     this.eventStore = eventStore;
     this.clock = clock;
+    this.alarmManager = (AlarmManager) context.getSystemService(Context.ALARM_SERVICE);
+    ;
+  }
+
+  @RestrictTo(RestrictTo.Scope.TESTS)
+  @VisibleForTesting
+  AlarmManagerScheduler(
+      Context applicationContext, EventStore eventStore, Clock clock, AlarmManager alarmManager) {
+    this.context = applicationContext;
+    this.eventStore = eventStore;
+    this.clock = clock;
+    this.alarmManager = alarmManager;
   }
 
+  @VisibleForTesting
+  boolean isJobServiceOn(Intent intent) {
+    return (PendingIntent.getBroadcast(context, 0, intent, PendingIntent.FLAG_NO_CREATE) != null);
+  }
+
+  /**
+   * Schedules the AlarmManager service.
+   *
+   * @param backendName The backend to where the events are logged.
+   * @param attemptNumber Number of times the AlarmManager has tried to log for this backend.
+   */
   @Override
-  public void schedule(TransportContext transportContext, int attemptNumber) {}
+  public void schedule(TransportContext transportContext, int attemptNumber) {
+    // Create Intent
+    String backendName = transportContext.getBackendName();
+    Uri.Builder intentDataBuilder = new Uri.Builder();
+    intentDataBuilder.appendQueryParameter(SchedulerUtil.BACKEND_NAME, backendName);
+    intentDataBuilder.appendQueryParameter(
+        SchedulerUtil.APPLICATION_BUNDLE_ID, BuildConfig.APPLICATION_ID);
+    Intent intent = new Intent(context, AlarmManagerScheduler.class);
```

Comment:
This does not look right, the scheduler is not a broadcast receiver, I bet it will crash at runtime, did you mean `AlarmManagerSchedulerService`?

Reply:
Updated